src/api/entities/guild/Guild.ts

```diff
@@ -89,12 +110,129 @@ export class Guild extends AbstractViewableGuild {
+
+    return this.edit({ systemChannelId }, reason)
+  }
+
+  async setRulesChannel(rulesChannel: GuildChannelResolvable, reason?: string): Promise<this | undefined> {
+    const rulesChannelId = resolveChannelId(rulesChannel)
+    if (!rulesChannelId) throw new DiscordooError('Guild#setRulesChannel', 'Cannot set rules channel to unknown channel.')
+
+    return this.edit({ rulesChannelId }, reason)
+  }
+
+  async setPublicUpdatesChannel(publicUpdatesChannel: GuildChannelResolvable, reason?: string): Promise<this | undefined> {
+    const publicUpdatesChannelId = resolveChannelId(publicUpdatesChannel)
+    if (!publicUpdatesChannelId) {
+      throw new DiscordooError('Guild#setPublicUpdatesChannel', 'Cannot set public updates channel to unknown channel.')
+    }
+
+    return this.edit({ publicUpdatesChannelId }, reason)
+  }
+
+  async setSafetyAlertsChannel(safetyAlertsChannel: GuildChannelResolvable, reason?: string): Promise<this | undefined> {
+    const safetyAlertsChannelId = resolveChannelId(safetyAlertsChannel)
+    if (!safetyAlertsChannelId) {
+      throw new DiscordooError('Guild#setSafetyAlertChannel', 'Cannot set safety alert channel to unknown channel.')
+    }
+
+    return this.edit({ safetyAlertsChannelId }, reason)
+  }
+
+  async setVerificationLevel(verificationLevel: GuildVerificationLevels, reason?: string): Promise<this | undefined> {
+    return this.edit({ verificationLevel }, reason)
+  }
+
+  async setDefaultNotifications(defaultNotifications: GuildDefaultMessageNotificationLevel, reason?: string): Promise<this | undefined> {
+    return this.edit({ defaultNotifications }, reason)
+  }
+
+  async setExplicitContentFilter(explicitContentFilter: GuildExplicitContentFilterLevels, reason?: string): Promise<this | undefined> {
+    return this.edit({ explicitContentFilter }, reason)
+  }
+
+  async setPremiumProgressBarEnabled(enabled: boolean, reason?: string): Promise<this | undefined> {
+    return this.edit({ premiumProgressEnabled: enabled }, reason)
+  }
+
+  async setIcon(icon: BufferResolvable, reason?: string): Promise<this | undefined> {
+    return this.edit({ icon }, reason)
+  }
+
+  async setSplash(splash: BufferResolvable, reason?: string): Promise<this | undefined> {
+    return this.edit({ splash }, reason)
+  }
+
+  async setBanner(banner: BufferResolvable, reason?: string): Promise<this | undefined> {
+    return this.edit({ banner }, reason)
+  }
+
+  async setWidgetEnabled(enabled: boolean, reason?: string): Promise<this | undefined> {
+    return this.edit({ widgetEnabled: enabled }, reason)
+  }
+
+  async setWidgetChannel(widgetChannel: GuildChannelResolvable, reason?: string): Promise<this | undefined> {
+    const widgetChannelId = resolveChannelId(widgetChannel)
+    if (!widgetChannelId) throw new DiscordooError('Guild#setWidgetChannel', 'Cannot set widget channel to unknown channel.')
```

Comment:
please rename all errors in this file to the general format: "Cannot do X without id."